### DISCUSSION TITLE
OMI glTF Working Group Meeting 2022-10-27

### DISCUSSION THREAD

**Discussion by robertlong** [omigroup/gltf-extensions#113] [General]
This meeting will be on 10/27/2022 at 10:30 PM (UTC) / 2:30 PM (PST) in the [OMI Discord](https://discord.gg/NJtT9grz5E) within the #omi-weekly-meeting channel. During the meeting, we will be using the #omi-gltf-extensions channel to manage a speaker queue, post links, and for any sidebar discussions.

To be notified of this meeting and others, subscribe to the [OMI Meetings and Events Calendar](https://calendar.google.com/calendar/u/1?cid=Y18wZHB1Z2Y5ZjgzZXE0cWVrbWI2b21xYmptZ0Bncm91cC5jYWxlbmRhci5nb29nbGUuY29t) or add yourself to the @omi-gltf-subgroup role in the #roles channel of the OMI Discord.

## Last Week's Agenda for reference:

- OMI_link
- OMI_spawn_point
- We're meeting in a Frame space, check the omi-gltf-extensions channel for details
- glTF Transform Updates
- [unity-thirdroom-exporter](https://github.com/matrix-org/thirdroom-unity-exporter)
- [OMI_collider](https://github.com/omigroup/gltf-extensions/pull/63)
- [KHR_audio / Khronos feedback](https://github.com/KhronosGroup/glTF/pull/2137)
  - [gltf-sample-assets](https://github.com/KhronosGroup/glTF-Sample-Models/pull/360)
- [glXF](https://github.com/KhronosGroup/glXF/blob/main/specification/2.0/README.md)
  - OMI_id / [OMI_ref](https://github.com/omigroup/gltf-extensions/pull/85)
- USDz spike
- Schedule alternate timezone meeting
- Show and tell
- Create next weeks agenda

To propose another item for the agenda, comment below (preferably before the meeting).

**Comment by robertlong** [comment]
# OMI_link

Link extension should be added to a node with the OMI_collider extension. Implementation details of how the link behaves (navigate on click or navigate on collide) is up to the client.

Metadata for the link can be gathered through a WebFinger-like API where you pass the URL and receive metadata like thumbnail, title, and description.

```json
{
  "nodes": [
    {
      "name": "Link",
      "extensions": {
        "OMI_link": {
          "uri": "https://omigroup.org/worlds/meetup"
        },
        "OMI_collider": {
          ...
        }
      }
    }
  ]
}
```

**Comment by madjin** [comment]
Visualizing an example when talking about trigger zones - can add a "zone" app in hyperfy that can trigger an action like opening a link in same tab / new tab over anything. Previews as a slightly transparent box overlay.

![brave_kFZNYGLz0i](https://user-images.githubusercontent.com/32600939/198407666-4d9bd58f-7311-476a-8646-729e4af42a3e.png)

Can also make any object into a clickable link, mouse cursor will change when an object is interactive.
![brave_atnAE8qUB4](https://user-images.githubusercontent.com/32600939/198407771-2d82dbea-1c61-4a03-b510-c35e0dde6430.png)

---

> Metadata for the link can be gathered through a WebFinger-like API where you pass the URL and receive metadata like thumbnail, title, and description.

Shared a couple examples with Wikipedia / gwern.net for hover text
![brave_scwVmgDJGK](https://user-images.githubusercontent.com/32600939/198407880-209fd04f-3644-4ff9-bc13-1a2399fd1b2e.png)
![brave_kEyVBaMwMS](https://user-images.githubusercontent.com/32600939/198407885-bbfe4ff9-ed31-49c5-8bcb-6d869012ceb0.png)

**Comment by robertlong** [comment]
# OMI_collider
Added some new comments on referring to colliders via indices vs placing definitions on nodes:
https://github.com/omigroup/gltf-extensions/pull/63#issuecomment-1294167810

**Comment by robertlong** [comment]
# three-omi
Discussed some implementation details for this PR, implementing KHR_audio https://github.com/omigroup/three-omi/pull/10
